src/Microsoft.Framework.Caching.Abstractions/CacheExtensions.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.Framework.Caching
+{
+    public static class CacheExtensions
+    {
+        /// <summary>
+        /// Sets the priority for keeping the cache entry in the cache during a memory pressure triggered cleanup.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="priority"></param>
+        public static CacheEntryOptions SetPriority(this CacheEntryOptions options, CacheItemPriority priority)
+        {
+            options.Priority = priority;
+            return options;
+        }
+
+        /// <summary>
+        /// Expire the cache entry if the given event occurs.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="trigger"></param>
+        public static CacheEntryOptions AddExpirationTrigger(
+            this CacheEntryOptions options,
+            [NotNull] IExpirationTrigger trigger)
+        {
+            options.Triggers.Add(trigger);
+            return options;
+        }
+
+        /// <summary>
+        /// Sets an absolute expiration time, relative to now.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="relative"></param>
+        public static CacheEntryOptions SetAbsoluteExpiration(this CacheEntryOptions options, TimeSpan relative)
+        {
+            options.AbsoluteExpirationRelativeToNow = relative;
+            return options;
+        }
+
+        /// <summary>
+        /// Sets an absolute expiration date for the cache entry.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="absolute"></param>
+        public static CacheEntryOptions SetAbsoluteExpiration(this CacheEntryOptions options, DateTimeOffset absolute)
+        {
+            if (options.AbsoluteExpiration == null)
+            {
+                options.AbsoluteExpiration = absolute;
+            }
+            else if (absolute < options.AbsoluteExpiration.Value)
```

Comment:
We can probably remove this check about checking the existing value. Just overwrite it.